src/app/AppLayout/AppLayout.tsx

```diff
@@ -152,6 +154,26 @@ const AppLayout: React.FunctionComponent<IAppLayout> = ({children}) => {
   const handleAboutModalToggle = () => {
     setAboutModalOpen(!aboutModalOpen);
   };
+
+  // TODO refactor
+  React.useEffect(() => {
+    const token = sessionStorage.getItem('token');
+    if(token) {
+      setShowUserInfo(true);
+    }
+  }, [serviceContext.api]);
+
+  const handleUserInfoToggle = () => {
+    setShowUserInfoDropdown(!showUserInfoDropdown);
+  }
+
+  const handleLogout = () => {
+    sessionStorage.removeItem('token');
+    setShowUserInfo(false);
+  };
```

Comment:
When "logout" gets clicked, the token is removed from session storage and the page does not re-render. When I manually refresh the page, another `useEffect` hook will recognize there are no stored credentials and bring me to the login page. How could we tell react to check the authorization  / re-render the page?

Reply:
The `routes.tsx` component is the one that "knows" if the user is authenticated and displays the Login component if not. `AppLayout` is the right place for the user menu stuff you've added here.